src/main/java/com/dev/voucherproject/model/storage/customer/CustomerDao.java

```diff
@@ -0,0 +1,144 @@
+
+import java.nio.ByteBuffer;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+
+@Repository
+public class CustomerDao implements CustomerStorage {
+    private static final Logger logger = LoggerFactory.getLogger(CustomerStorage.class);
+
+    public static final String INSERT = "INSERT INTO customers(customer_id, name, email, created_at) " +
+        "VALUES (UUID_TO_BIN(:customerId), :name, :email, :createdAt)";
+
+    public static final String FIND_ALL = "SELECT * FROM customers";
+
+    public static final String FIND_BY_UUID = "select * from customers where customer_id = UUID_TO_BIN(:customerId)";
+
+    public static final String DELETE_ALL = "DELETE FROM customers";
+
+    public static final String FIND_BY_NAME = "select * from customers where name = :name";
+
+    public static final String FIND_BY_EMAIL = "select * from customers where email = :email";
+    public static final String DELETE_BY_UUID = "DELETE FROM customers WHERE customer_id = UUID_TO_BIN(:customerId)";
+
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    public CustomerDao(NamedParameterJdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private static final RowMapper<Customer> rowMapper = (resultSet, i) -> {
+        String customerName = resultSet.getString("name");
+        String email = resultSet.getString("email");
+        UUID customerId = toUUID(resultSet.getBytes("customer_id"));
+        LocalDateTime createdAt = resultSet.getTimestamp("created_at").toLocalDateTime();
+
+        return new Customer(customerId, customerName, email, createdAt);
+    };
+
+    private SqlParameterSource toParamMap(Customer customer) {
+        return new MapSqlParameterSource()
+            .addValue("customerId", customer.getCustomerId().toString().getBytes())
+            .addValue("name", customer.getName())
+            .addValue("email", customer.getEmail())
+            .addValue("createdAt", Timestamp.valueOf(customer.getCreatedAt()));
+    }
+
+    @Override
+    public void insert(Customer customer) {
+        try {
+            jdbcTemplate.update(INSERT, toParamMap(customer));
+        } catch (DuplicateKeyException e) {
+            logger.warn("{} 는 이미 존재하는 이메일입니다.", customer.getEmail());
+        }
+    }
+
+    @Override
+    public List<Customer> findAll() {
+        return jdbcTemplate.query(FIND_ALL, rowMapper);
+    }
+
+    @Override
+    public Optional<Customer> findById(UUID customerId) {
+        try {
+            return Optional.ofNullable(
+                jdbcTemplate.queryForObject(FIND_BY_UUID, Collections.singletonMap("customerId", customerId.toString().getBytes()), rowMapper)
+            );
+        } catch (EmptyResultDataAccessException e) {
+            logger.warn("{} 는 존재하지 않는 고객 아이디입니다.", customerId);
+        }
+        return Optional.empty();
```

Comment:
```suggestion
    public Optional<Customer> findById(UUID customerId) {
        try {
            return Optional.ofNullable(
                jdbcTemplate.queryForObject(FIND_BY_UUID, Collections.singletonMap("customerId", customerId.toString().getBytes()), rowMapper)
            );
        } catch (EmptyResultDataAccessException e) {
           return Optional.empty();
        }
```
대체값을 리턴하는데 로그를 남길 필요가 없어보여요!